cms/djangoapps/contentstore/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,168 @@
+                    verified_course_user_group_partition_group.partition_id,
+                    None
+                )
+                if (verified_partition_group_access
+                        and verified_course_user_group_partition_group.group_id in verified_partition_group_access):
+                    set_verified_enrollment_track = True
+
+                # Add the enrollment track ids to a group access array
+                enrollment_track_group_access = []
+                if set_audit_enrollment_track:
+                    enrollment_track_group_access.append(audit_course_mode.id)
+                if set_verified_enrollment_track:
+                    enrollment_track_group_access.append(verified_course_mode.id)
+
+                # If either the audit track, or verified track needed to be updated set the access, update and publish
+                if set_verified_enrollment_track or set_audit_enrollment_track:
+                    # Check that the xblock is published if it needs changes, otherwise raise an error
+                    if not published:
+                        raise CommandError('XBlock for course, %s needs access changes, but is a draft' % course_id)
+                    item.group_access = {ENROLLMENT_TRACK_PARTITION_ID: enrollment_track_group_access}
+                    module_store.update_item(item, ModuleStoreEnum.UserID.mgmt_command)
+                    module_store.publish(item.location, ModuleStoreEnum.UserID.mgmt_command)
+
+        partitions_to_delete = random_audit_course_user_group_partition_groups
+        partitions_to_delete.append(verified_course_user_group_partition_group)
+
+        # Check if we should delete any partition groups,
+        # Taken from contentstore/views/course.py.remove_content_or_experiment_group
+        if partitions_to_delete:
+            partition_service = PartitionService(course_key)
+            course = partition_service.get_course()
+            for partition_to_delete in partitions_to_delete:
+                # Get the user partition, and the index of that partition in the course
+                partition = partition_service.get_user_partition(partition_to_delete.partition_id)
+                partition_index = course.user_partitions.index(partition)
+                group_id = int(partition_to_delete.group_id)
+
+                # Sanity check to verify that all of the groups being deleted are empty,
+                # since they should have been converted to use enrollment tracks instead
+                usages = GroupConfiguration.get_partitions_usage_info(module_store, course)
+                used = group_id in usages
+                if used:
+                    raise("Content group %s is in use and cannot be deleted." % partition_to_delete.group_id)
+
+                # Remove the groups that match the group ID of the partition to be deleted
+                matching_groups = [group for group in partition.groups if group.id == group_id]
```

Comment:
I would expect us to be looking within a particular partition, and then removing groups within that. If all groups are removed in the partition, we should delete the partition entirely.